### ISSUE TITLE
SMPLX fitting

### ISSUE THREAD

**Issue by Jun-Pu** [SiTH-Diffusion/SiTH#10]
Thanks so much for the inspiring work!

I would like to know if there are any methods to improve the SMPLX estimation results? 

I tested with my own data (shown below) and found that the SMPLX matching result is not satisfactory. I'm not quite sure of the specific reasons and don't know how to improve it.

Could you please help me to have a look? Thanks in advance！

![20240804111852](https://github.com/user-attachments/assets/b826a8ff-0912-40a6-8888-fc9d0a7495b6)

**Comment by Jun-Pu** [comment]
problem solved, thx

**Comment by azuxmioy** [comment]
Hi,

Sorry for the late reply. This is a very good question.

Since SMPL-X fitting is really case-by-case, there is no perfect parameter for all datasets. I'd like to share some tips when I do body fitting.

1. Make your subject a bit larger. As the image is 2 meters tall, a human body around 1.6~1.8 meters better fits the distribution of betas. 

2. Use the opt_pose option to optimize pose parameters. While this minimizes the 2D keypoint projection errors, it may also introduce weird limb bending in 3D. 

3. There are some hyperparameters you can play with. For example, if you want to improve arm poses not leg poses you need to optimize the [skeleton ID](https://github.com/SiTH-Diffusion/SiTH/blob/main/fit.py#L58) of [14, 15, 16, 17, 18, 19, 20]. If the 2D keypoint errors are still too large, you can adjust the [learning rate](https://github.com/SiTH-Diffusion/SiTH/blob/main/fit.py#L54) and increase the optimization iterations. 

I hope this information helps!

**Comment by Jun-Pu** [comment]
> Hi,
> 
> Sorry for the late reply. This is a very good question.
> 
> Since SMPL-X fitting is really case-by-case, there is no perfect parameter for all datasets. I'd like to share some tips when I do body fitting.
> 
> 1. Make your subject a bit larger. As the image is 2 meters tall, a human body around 1.6~1.8 meters better fits the distribution of betas.
> 2. Use the opt_pose option to optimize pose parameters. While this minimizes the 2D keypoint projection errors, it may also introduce weird limb bending in 3D.
> 3. There are some hyperparameters you can play with. For example, if you want to improve arm poses not leg poses you need to optimize the [skeleton ID](https://github.com/SiTH-Diffusion/SiTH/blob/main/fit.py#L58) of [14, 15, 16, 17, 18, 19, 20]. If the 2D keypoint errors are still too large, you can adjust the [learning rate](https://github.com/SiTH-Diffusion/SiTH/blob/main/fit.py#L54) and increase the optimization iterations.
> 
> I hope this information helps!

Awesome! Thanks so much!
